stripe/_api_requestor.py

```diff
@@ -201,6 +201,7 @@ def request(
             params=params,
             requestor=requestor,
             api_mode=api_mode,
+            v2_deleted_object=method == "delete" and api_mode == "V2",
```

Comment:
this is the runtime code that ensures we turn delete responses into `V2DeletedObject` instead of `EventDestination`

We're not merging this second, but it's what we'll do when we _are_ ready to merge.